From 70ef06c844ba112020e548691bffb2b8da70602b Mon Sep 17 00:00:00 2001
From: Alfred Chen <cchalpha@gmail.com>
Date: Wed, 8 Apr 2020 22:00:31 +0800
Subject: [PATCH 1/2] bmq: Fix sync-up error when BMQ applied but not enabled.

This should be apply to c36a1f520471 BMQ 0.97 in next kernel release.
---
 kernel/sched/sched.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/kernel/sched/sched.h b/kernel/sched/sched.h
index 911b30506c83..f38d1343e2bf 100644
--- a/kernel/sched/sched.h
+++ b/kernel/sched/sched.h
@@ -2484,6 +2484,19 @@ static inline void membarrier_switch_mm(struct rq *rq,
 }
 #endif
 
+#ifdef CONFIG_SMP
+static inline bool is_per_cpu_kthread(struct task_struct *p)
+{
+	if (!(p->flags & PF_KTHREAD))
+		return false;
+
+	if (p->nr_cpus_allowed != 1)
+		return false;
+
+	return true;
+}
+#endif
+
 static inline int task_running_nice(struct task_struct *p)
 {
 	return (task_nice(p) > 0);
-- 
2.26.0.106.g9fadedd637

